apollo-federation/src/operation/mod.rs

```diff
@@ -3648,6 +3616,397 @@ impl NamedFragments {
     }
 }
 
+// @defer handling
+
+const DEFER_DIRECTIVE_NAME: &str = "defer";
+
+#[derive(Debug, Clone, Copy)]
+enum DeferFilter<'a> {
+    All,
+    Labels(&'a IndexSet<String>),
+}
+
+impl DeferFilter<'_> {
+    fn remove_defer(&self, directive_list: &mut DirectiveList, schema: &apollo_compiler::Schema) {
+        match self {
+            Self::All => {
+                directive_list.remove_one(DEFER_DIRECTIVE_NAME);
+            }
+            Self::Labels(set) => {
+                let label = directive_list
+                    .get(DEFER_DIRECTIVE_NAME)
+                    .and_then(|directive| directive.argument_by_name("label", schema).ok())
```

Comment:
nit: since we have `DEFER_DIRECTIVE_NAME` as a const -> should `label` also be a const?